atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/PostgresConnectionConfig.java

```diff
@@ -87,6 +87,7 @@ public Properties getHikariProperties() {
         props.setProperty("password", getDbPassword().unmasked());
 
         props.setProperty("tcpKeepAlive", "true");
+        props.setProperty("targetServerType", "primary");
```

Comment:
> `targetServerType` (String) Default `any`
> Allows opening connections to only servers with required state, the allowed values are `any`, `primary`, `master`, `slave`, `secondary`, `preferSlave`, `preferSecondary` and `preferPrimary`. The `primary`/`secondary` distinction is currently done by observing if the server allows writes. The value `preferSecondary` tries to connect to secondary if any are available, otherwise allows falls back to connecting also to primary. The value `preferPrimary` tries to connect to primary if it is available, otherwise allows falls back to connecting to secondaries available.